### PR TITLE
[MM-44757] Restrict runs and playbooks in LHS

### PR DESCRIPTION
**Pull request by iomodo**
#### Summary
This PR Restricts runs and playbooks

#### Ticket Link
  Fixes: 
https://mattermost.atlassian.net/browse/MM-44757
https://mattermost.atlassian.net/browse/MM-44758

#### Checklist

- [ ] Telemetry updated
- [ ] Gated by experimental feature flag
- [ ] Unit tests updated

### CODE REVIEW

File: server/api/categories.go [hunk 1]
```diff
@@ -45,7 +52,25 @@ func (h *CategoryHandler) getMyCategories(w http.ResponseWriter, r *http.Request
...
 		return
 	}
-	ReturnJSON(w, categories, http.StatusOK)
+
+	runsCategory, err := h.getRunsCategory(teamID, userID)
+	if err != nil {
+		h.HandleError(w, err)
+		return
+	}
+	filteredRuns := filterItemsInCategory(runsCategory, categories)
```
Comment (crspeller) [inline server/api/categories.go:61]: What is this about? This seems quite inefficient and isn't it already done by the DB?

File: webapp/src/components/sidebar/playbooks_sidebar.tsx [hunk 2]
```diff
@@ -1,115 +1,71 @@
...
-    const runsItems = playbookRuns ? playbookRuns.map((run): GroupItem => {
-        return {
-            areaLabel: run.name,
-            className: '',
-            display_name: run.name,
-            icon: <StyledPlaybookRunIcon/>,
-            isCollapsed: false,
-            itemMenu: null,
-            link: pluginUrl(`/run_details/${run.id}`),
+    useEffect(() => {
```
Comment (crspeller) [inline webapp/src/components/sidebar/playbooks_sidebar.tsx:53]: Could we use a custom hook here to make it more clear?

File: server/app/category.go [hunk 3]
```diff
@@ -59,6 +61,15 @@ func (c *Category) IsValid() error {
 	return nil
 }
 
+func (c *Category) ContainsItem(item CategoryItem) bool {
+	for _, catItem := range c.Items {
+		if catItem.ItemID == item.ItemID && catItem.Type == item.Type {
```
[the remark this answers is not in the document]
Comment (iomodo) [reply server/app/category.go:66]: Theoretically yes. ItemID is either playbookID or runID and they can be the same(low probability but still)

File: server/api/categories.go [hunk 4]
```diff
@@ -8,21 +8,28 @@ import (
 	"github.com/gorilla/mux"
 	"github.com/mattermost/mattermost-plugin-playbooks/server/app"
 	"github.com/mattermost/mattermost-plugin-playbooks/server/bot"
+	"github.com/pkg/errors"
 
 	pluginapi "github.com/mattermost/mattermost-plugin-api"
 )
 
+const maxItemsInRunsAndPlaybooksCategory = 100
```
[the remark this answers is not in the document]
Comment (iomodo) [reply server/api/categories.go:16]: Should we have no limit here? I'm 0/5

File: server/api/categories.go (hunk 1 above)
Comment (crspeller) [reply server/api/categories.go:61]: I think the function and variable names here are a bit confusing as to what is happening. 
`filterItemsInCategory` -> `filterDuplicatesFromCategories`? 
`categories` -> `userCategories` or `customCategories`

**Review by crspeller — CHANGES_REQUESTED** [review]
Existing comments.

**Review by crspeller — APPROVED** [review]
Unblocking for after the limit change.

**Review by calebroseland — APPROVED** [review]

**Pull request merged**